pkg/command/service/scale..go

```diff
@@ -0,0 +1,295 @@
+	if err != nil {
+		fmt.Printf("failed to generate json file and skip %s\n", err)
+	}
+	fmt.Printf("Measurement saved in JSON file %s\n", jsonPath)
+
+	htmlPath := filepath.Join(outputLocation, fmt.Sprintf("%s_%s", current.Format(DateFormatString), "ksvc_creation_time.html"))
+	err = utils.GenerateHTMLFile(csvPath, htmlPath)
+	if err != nil {
+		fmt.Printf("failed to generate HTML file and skip %s\n", err)
+	}
+	fmt.Printf("Visualized measurement saved in HTML file %s\n", htmlPath)
+
+	return nil
+}
+
+func scaleAndMeasure(params *pkg.PerfParams, inputs pkg.ScaleArgs, nsNameList []string, servicesListFunc func(servingv1client.ServingV1Interface, []string, string) []ServicesToScale) (pkg.ScaleResult, error) {
+	result := pkg.ScaleResult{}
+	ksvcClient, err := params.NewServingClient()
+	if err != nil {
+		return result, err
+	}
+	objs := servicesListFunc(ksvcClient, nsNameList, inputs.SvcPrefix)
+	count := len(objs)
+
+	var wg sync.WaitGroup
+	wg.Add(count)
+	for i := 0; i < count; i++ {
+		ndx := i
+		go func() {
+			defer wg.Done()
+			sdur, ddur, err := runScaleFromZero(params, objs[ndx].Namespace, objs[ndx].Service)
+			if err == nil {
+				//measure
+				fmt.Printf("result of scale for service %s is %f, %f \n", objs[ndx].Service.Name, sdur.Seconds(), ddur.Seconds())
+				result.Measurment = append(result.Measurment, pkg.ScaleFromZeroResult{
+					ServiceName:       objs[ndx].Service.Name,
+					ServiceNamespace:  objs[ndx].Service.Namespace,
+					ServiceLatency:    sdur.Seconds(),
+					DeploymentLatency: ddur.Seconds(),
+				})
+			} else {
+				fmt.Printf("result of scale is error: %s", err)
+			}
+		}()
+	}
+	wg.Wait()
+
+	return result, nil
+}
+
+func getServices(servingClient servingv1client.ServingV1Interface, nsNameList []string, svcPrefix string) []ServicesToScale {
+	objs := []ServicesToScale{}
+	for _, ns := range nsNameList {
+		svcList, err := servingClient.Services(ns).List(context.TODO(), metav1.ListOptions{})
+		if err == nil {
+			for _, s := range svcList.Items {
+				svc := s
+				if strings.HasPrefix(s.Name, svcPrefix) {
+					objs = append(objs, ServicesToScale{Namespace: ns, Service: &svc})
+				}
+			}
+		}
+	}
+	return objs
+}
+
+func runScaleFromZero(params *pkg.PerfParams, namespace string, svc *servingv1.Service) (
+	time.Duration, time.Duration, error) {
+	selector := labels.SelectorFromSet(labels.Set{
+		serving.ServiceLabelKey: svc.Name,
+	})
+
+	watcher, err := params.ClientSet.AppsV1().Deployments(namespace).Watch(
+		context.Background(), metav1.ListOptions{LabelSelector: selector.String()})
+	if err != nil {
+		m := fmt.Sprintf("unable to watch the deployment for the service: %v", err)
+		log.Println(m)
+		return 0, 0, errors.New(m)
+	}
+	defer watcher.Stop()
+
+	ddch := watcher.ResultChan()
+	sdch := make(chan struct{})
+	errch := make(chan error)
+
+	url := svc.Status.RouteStatusFields.URL.URL()
+
+	go func() {
+		_, err = Poll(10, 2*time.Second, 2*time.Second, url.String())
```

Comment:
Hi @nader-ziada ,

Not sure what is your scale scenario, I'm thinking maybe it's good to make this configurable, the requestInterval and requestTimeout, so that people could test scale under different workloads, WDYT?

Reply:
you mean. the intervals? yeah will make that configurable